File: app/src/main/java/uk/co/sundroid/activity/data/DataGroup.java
package uk.co.sundroid.activity.data;

public enum DataGroup {

	DAY_SUMMARY ("Day summary", 0),
	DAY_DETAIL ("Day in detail", 1),
    TRACKER ("Sun and moon tracker", 2),
    MONTH_CALENDARS ("Month calendars", 3),
    MONTH_MOONPHASE ("Moon phase calendar", 4),
    YEAR_EVENTS ("Year events", 5);

	private final int index;
	
	private final String name;
	
	DataGroup(String name, int index) {
		this.name = name;
		this.index = index;
	}

	public int getIndex() {
		return index;
	}

	public String getName() {
		return name;
	}
	
	public static DataGroup forIndex(int index) {
		for (DataGroup dataView : values()) {
			if (dataView.getIndex() == index) {
				return dataView;
			}
		}
		return null;
	}
	
}

File: app/src/main/kotlin/uk/co/sundroid/activity/data/DataGroup.kt
package uk.co.sundroid.activity.data

import android.app.Fragment
import uk.co.sundroid.R
import uk.co.sundroid.activity.data.fragments.*

enum class DataGroup constructor(val displayName: String, val fragmentClass: Class<out Fragment>) {

    DAY_DETAIL("Day in detail", DayDetailFragment::class.java),
    TRACKER("Sun and moon tracker", TrackerFragment::class.java),
    MONTH_CALENDARS("Month calendars", MonthCalendarsFragment::class.java),
    MONTH_MOONPHASE("Moon phase calendar", MonthMoonPhaseFragment::class.java),
    YEAR_EVENTS("Year events", YearEventsFragment::class.java);

}
